src/components/v5/common/ActionSidebar/partials/forms/PaymentBuilderForm/hooks.ts

```diff
@@ -123,8 +132,27 @@ export const useValidationSchema = () => {
             .defined()
             .required(),
         })
+        .test(
+          'is-in-colony',
+          formatText({ id: 'actionSidebar.tokenAddress.error' }),
+          (item) => {
+            const { payments } = item || {};
+
+            if (!payments) {
+              return false;
+            }
+
+            return payments.every((payment) => {
+              return colonyTokens.some(
+                (colonyToken) =>
+                  colonyToken.tokenAddress === payment.tokenAddress,
+              );
+            });
+          },
+        )
         .defined()
         .concat(ACTION_BASE_VALIDATION_SCHEMA),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Is this preferred over wrapping `colonyTokens` in its own `useMemo`? We probably want to re-run the validations when tokens change.